### PR TITLE
feat(compute): Firewall samples refactor

### PR DESCRIPTION
**Pull request by rsamborski**
Moving firewall samples to a different folder to align with folder structure agreed for future compute samples development.

### CODE REVIEW

File: compute/cloud-client/firewall/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+1. **Install dependencies** via [Composer](http://getcomposer.org/doc/00-intro.md).
+    Run `php composer.phar install` (if composer is installed locally) or `composer install`
+    (if composer is installed globally).
+
+1. Create a service account at the
+[Service account section in the Cloud Console](https://console.cloud.google.com/iam-admin/serviceaccounts/)
+
+1. Download the json key file of the service account.
+
+1. Set `GOOGLE_APPLICATION_CREDENTIALS` environment variable pointing to that file.
```
Comment (savijatv) [inline compute/cloud-client/firewall/README.md:54]: Set the 'GOOGLE_APPLICATION....

File: compute/cloud-client/firewall/README.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+## Install Dependencies
+
+1. **Install dependencies** via [Composer](http://getcomposer.org/doc/00-intro.md).
+    Run `php composer.phar install` (if composer is installed locally) or `composer install`
+    (if composer is installed globally).
+
+1. Create a service account at the
+[Service account section in the Cloud Console](https://console.cloud.google.com/iam-admin/serviceaccounts/)
+
+1. Download the json key file of the service account.
```
Comment (savijatv) [inline compute/cloud-client/firewall/README.md:52]: Maybe a link here helps - https://cloud.google.com/iam/docs/creating-managing-service-account-keys#getting_a_service_account_key

File: compute/cloud-client/firewall/README.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+1. Set `GOOGLE_APPLICATION_CREDENTIALS` environment variable pointing to that file.
+
+## Samples
+
+To run the Compute Samples, run any of the files in `src/` on the CLI to print
```
Comment (savijatv) [inline compute/cloud-client/firewall/README.md:58]: To run the Compute samples, ....

File: compute/cloud-client/firewall/README.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
+Google Cloud Compute PHP Instances Samples
```
[the remark this answers is not in the document]
Comment (rsamborski) [reply compute/cloud-client/firewall/README.md:1]: I am changing to: 
 - "Google Cloud Compute Engine PHP Samples - Instances" in instances folder
 - "Google Cloud Compute Engine PHP Samples - Firewall" in firewall folder

File: compute/cloud-client/firewall/README.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+1. When running locally, use the [Google Cloud SDK][google-cloud-sdk]
+
+        gcloud auth application-default login
+
+1. When running on App Engine or Compute Engine, credentials are already
+   set-up. However, you may need to configure your Compute Engine instance
+   with [additional scopes][additional_scopes].
+
+1. You can create a [Service Account key file][service_account_key_file]. This file can be used to
+   authenticate to Google Cloud Platform services from any environment. To use
```
[the remark this answers is not in the document]
Comment (rsamborski) [reply compute/cloud-client/firewall/README.md:33]: Sorry, but I don't agree with this suggestion. We are authenticating our calls **to** Google Cloud Platform services/APIs. Removing 'to' means we run the platform services and authenticate them to some other system.

File: compute/cloud-client/firewall/README.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+[additional_scopes]: https://cloud.google.com/compute/docs/authentication#using
+[service_account_key_file]: https://developers.google.com/identity/protocols/OAuth2ServiceAccount#creatinganaccount
+
+## Install Dependencies
+
+1. **Install dependencies** via [Composer](http://getcomposer.org/doc/00-intro.md).
+    Run `php composer.phar install` (if composer is installed locally) or `composer install`
+    (if composer is installed globally).
+
+1. Create a service account at the
```
[the remark this answers is not in the document]
Comment (rsamborski) [reply compute/cloud-client/firewall/README.md:49]: I am changing as suggested.

File: compute/cloud-client/firewall/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (savijatv) [reply compute/cloud-client/firewall/README.md:33]: The context is not clear. Does it mean that the key file is used to 'access' GCP services from any environment?

[the remark this answers is not in the document]
Comment (rsamborski) [reply compute/cloud-client/firewall/README.md:33]: Yes. You basically use the key to access GCP services from anywhere (local machine, hosted environment, other cloud, etc.)

**Review by bshaffer — APPROVED** [review]

File: compute/cloud-client/firewall/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (bshaffer) [reply compute/cloud-client/firewall/README.md:33]: I agree that "to" sounds a big clumsy here, and perhaps it should be "authenticate for" or "authenticate with"

[the remark this answers is not in the document]
Comment (rsamborski) [reply compute/cloud-client/firewall/README.md:33]: I have chatted with Alberto (TW in Compute) - he did a quick search in Code Search, and "to" is much more prevalent than "for":
- TO: http://shortn/_AijuPjE4J5
- FOR: http://shortn/_5x1FsF7det

Plus I also found this as an example of to - [GitHub Docs](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/about-authentication-to-github).

[the remark this answers is not in the document]
Comment (rsamborski) [reply compute/cloud-client/firewall/README.md:33]: Found one more twist here: https://developers.google.com/style/word-list#authentication-and-authorization
The doc suggests using `authenticate against`, which seems to work too in this case.

**Pull request merged**